### PR TITLE
chore(errors): retry setupPlugin on FetchError errors

### PR DESCRIPTION
**Pull request by hazzadous**
We occasionally see socket errors e.g. `socket hang up`. Here we ensure
that the caller of setupPlugin knows that it should retry on these
errors.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -50,85 +51,140 @@ export const setupPlugin: BigQueryPlugin['setupPlugin'] = async (meta) => {
...
     const credentials = JSON.parse(attachments.googleCloudKeyJson.contents.toString())
-    global.bigQueryClient = new BigQuery({
-        projectId: credentials['project_id'],
-        credentials,
-        autoRetry: false,
-    })
 
-    global.bigQueryTable = global.bigQueryClient.dataset(config.datasetId).table(config.tableId)
+    try {
+        global.bigQueryClient = new BigQuery({
```
Comment (tiina303) [inline index.ts:56]: does their constructor do any calls 🤷‍♀️  (just included in the try/catch to be safe)

File: index.ts [hunk 2]
```diff
@@ -50,85 +51,140 @@ export const setupPlugin: BigQueryPlugin['setupPlugin'] = async (meta) => {
...
+                    // a different worker already updated the table
+                    const fieldsToStillAdd = BIG_QUERY_TABLE_FIELDS.filter(
+                        ({ name }) => !result.schema?.fields?.find((f: any) => f.name === name)
                     )
+
+                    if (fieldsToStillAdd.length > 0) {
+                        console.error(`Tried adding fields ${JSON.stringify(fieldsToAdd)}, but ${JSON.stringify(
+                            fieldsToStillAdd
+                        )} still to add. Can not start plugin.`)
+                        throw error
```
Comment (tiina303) [inline index.ts:146]: this could also be FetchError 🤷‍♀️ , so should be retried

File: index.ts (hunk 1 above)
Comment (hazzadous) [reply index.ts:56]: Yeah not sure tbh, I think not but… we can refine later if neede

**Review by hazzadous — COMMENTED** [review]
Can’t approve as I’m the pr creator but looks good.

File: index.ts (hunk 2 above)
Comment (hazzadous) [reply index.ts:146]: That will bubble up to the top level try/catch though right? So would result in a retry anyway?

Comment (tiina303) [reply index.ts:146]: jup exactly I changed this as it previously created a custom error here (now just logging that part)

**Pull request merged**